sqlalchemy_jsonapi/unittests/test_serializer_post_collection.py

```diff
@@ -40,5 +83,208 @@ def test_201_resource_creation(self):
+
+        with self.assertRaises(errors.ValidationError) as error:
+            models.serializer.post_collection(
+                self.session, payload, 'users')
+
+        self.assertEqual(error.exception.status_code, 409)
+
+    def test_add_resource_mismatched_endpoint(self):
+        """Create resource with mismatched returns 409.
+
+        A InvalidTypeEndpointError is raised.
+        """
+        payload = {
+            'data': {
+                'type': 'posts'
+            }
+        }
+
+        with self.assertRaises(errors.InvalidTypeForEndpointError) as error:
+            models.serializer.post_collection(self.session, payload, 'users')
+
+        self.assertEqual(
+            error.exception.detail, 'Expected users, got posts')
+        self.assertEqual(error.exception.status_code, 409)
+
+    def test_add_resource_with_missing_data(self):
+        """Create resource with missing content data results in 400.
+
+        A BadRequestError is raised.
+        """
+        payload = {}
+
+        with self.assertRaises(errors.BadRequestError) as error:
+            models.serializer.post_collection(self.session, payload, 'users')
+
+        self.assertEqual(
+            error.exception.detail, 'Request should contain data key')
+        self.assertEqual(error.exception.status_code, 400)
+
+    def test_add_resource_with_missing_type(self):
+        """Creat resource without type results in 409.
+
+        A MissingTypeError is raised.
+        """
+        payload = {
+            'data': {
+                'attributes': {
+                    'first': 'Sally',
+                    'last': 'Smith',
+                    'username': 'SallySmith1',
+                    'password': 'password',
+                }
+            }
+        }
+
+        with self.assertRaises(errors.MissingTypeError) as error:
+            models.serializer.post_collection(self.session, payload, 'users')
+
+        self.assertEqual(
+            error.exception.detail, 'Missing /data/type key in request body')
+        self.assertEqual(error.exception.status_code, 409)
+
+    def test_add_resource_with_unknown_field_name(self):
+        """Create resource with unknown field results in 409.
+
+        A ValidationError is raised.
+        """
+        payload = {
+            'data': {
+                'type': 'users',
+                'attributes': {
+                    'first': 'Sally',
+                    'last': 'Smith',
+                    'username': 'SallySmith1',
+                    'password': 'password',
+                    'unknown-attribute': 'test'
+                }
+            }
+        }
+
+        with self.assertRaises(errors.ValidationError) as error:
+            models.serializer.post_collection(
+                self.session, payload, 'users')
+
+        self.assertEqual(error.exception.status_code, 409)
+
+# MISSING TEST FOR INVALID VALUE
+
+# MISSING TEST FOR PERMISSION DENIED
```

Comment:
Are these missing tests your next step? I don't mind leaving comments like this in if they're being resolved quickly, but we shouldn't leave them in there.